File: app/src/main/java/com/binarybricks/coinhood/network/schedulers/TestScheduleProvider.java
package com.binarybricks.coinhood.network.schedulers;

import android.support.annotation.NonNull;

import io.reactivex.schedulers.TestScheduler;

/**
 * Implementation of the {@link BaseSchedulerProvider} making all {@link io.reactivex.schedulers.TestScheduler} for testing purposes.
 */
public class TestScheduleProvider implements BaseSchedulerProvider {

    private TestScheduler scheduler;

    public TestScheduleProvider() {
        scheduler = new TestScheduler();
    }

    @NonNull
    @Override
    public TestScheduler computation() {
        return scheduler;
    }

    @NonNull
    @Override
    public TestScheduler io() {
        return scheduler;
    }

    @NonNull
    @Override
    public TestScheduler ui() {
        return scheduler;
    }
}

File: app/src/main/java/com/binarybricks/coiny/network/schedulers/TestScheduleProvider.kt
package com.binarybricks.coiny.network.schedulers

import io.reactivex.schedulers.TestScheduler

/**
 * Implementation of the [BaseSchedulerProvider]
 * making all [io.reactivex.schedulers.TestScheduler] for testing purposes.
 */
class TestScheduleProvider : BaseSchedulerProvider {

    private val scheduler: TestScheduler = TestScheduler()

    override fun computation(): TestScheduler {
        return scheduler
    }

    override fun io(): TestScheduler {
        return scheduler
    }

    override fun ui(): TestScheduler {
        return scheduler
    }
}
